src/entry.rs

```diff
@@ -358,54 +372,15 @@ impl<'a> EntryFields<'a> {
                                           name found"))
             };
 
-            // Ok, we're going to try to create a symlink. We need to protect
-            // against symlinks which point outside the destination root
-            // directory, however. Otherwise it could be possible to
-            // accidentally write files outside there with malformed tarballs.
-            //
-            // To do that we take a look at the link name for this target, `src`
-            // above. We then recreate the target that we're actually going to
-            // link to, `actual_src`. Root directories and the current directory
-            // are skipped (like `unpack_in` above) and `..` is allowed, but
-            // only if it doesn't escape the root directory. This should allow
-            // for relative symlinks within the destination but disallow
-            // symlinks that point outside.
-            let mut target = dst.to_path_buf();
-            target.pop();
-            let mut actual_src = PathBuf::new();
-            for part in src.components() {
-                match part {
-                    Component::Prefix(..) |
-                    Component::RootDir |
-                    Component::CurDir => continue,
-                    Component::ParentDir => {
-                        actual_src.push("..");
-                        if !target.pop() {
-                            return Err(other("symlink destination points \
-                                              outside unpack destination"))
-                        }
-                        if let Some(root) = root {
-                            if !target.starts_with(root) {
-                                return Err(other("symlink destination points \
-                                                  outside unpack destination"))
-                            }
-                        }
-                    }
-                    Component::Normal(part) => {
-                        target.push(part);
-                        actual_src.push(part);
-                    }
-                }
-            }
-            if actual_src.iter().count() == 0 {
+            if src.iter().count() == 0 {
                 return Err(other("symlink destination is empty"))
             }
 
-            println!("{:?} {:?}", actual_src, dst);
+            println!("{:?} {:?}", src, dst);
```

Comment:
Yes, I re-introduced it later in order to get some clues about the appveyor issue.